scene/gui/rich_text_label.cpp

```diff
@@ -884,7 +894,11 @@ void RichTextLabel::_update_scroll() {
 void RichTextLabel::_update_fx(RichTextLabel::ItemFrame *p_frame, float p_delta_time) {
 	Item *it = p_frame;
 	while (it) {
-		ItemFX *ifx = Object::cast_to<ItemFX>(it);
+		ItemFX *ifx = NULL;
```

Comment:
Be careful here. This branch of the code is used to provide shake with it's random direction per line -- NULL pointers are bypassing this and shake will behave not as intended.

Reply:
it's fine, because of the if at line 903. ItemShake is handled after that.

Reply:
Okay I pretty sure I misunderstood what you said first. So what do you mean by this?

Reply:
No actually, having looked at your branch more you're correct. It just wasn't super clear when I was previewing the diff.